task/host-sp-comms/src/main.rs

```diff
@@ -175,6 +179,12 @@ impl ServerImpl {
         }
     }
 
+    fn set_debug_impl(&mut self, debug: DebugReg) {
+        if debug != self.debug {
```

Comment:
Tiny nitpick - we don't need this conditional here; this can be just `self.debug = debug`. `set_status_impl()` has a check on whether its value changed to avoid needlessly calling into the `sys` task to set the GPIO to what it already is.